### PR TITLE
chore: rename `CometRowToColumnar` and fix duplication bug

### LINKED ISSUE #770: Rename `CometRowToColumnarExec`
### What is the problem the feature request solves?

https://github.com/apache/datafusion-comet/pull/731 updated `CometRowToColumnarExec` to accept columnar input as well, so the name "RowToColumnar" is a little misleading now. We should change this name and probably update the configs to match for enabling this.

### Describe the potential solution

Ideas:
- CometSparkToColumnarExec
- CometConvertFromSparkExec
- CometFromSparkExec
- CometColumnarFromSparkExec

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Kimahriman**
## Which issue does this PR close?

Closes #770 

## What changes are included in this PR?
Rename `CometRowToColumnar`  to `CometSparkToColumnar`. Open to other names, wanted to at least get the conversation started.

Additionally, I included a bug fix I encountered trying out `CometRowToColumnar` where AQE was inserting a duplicate node. I can move this into a separate PR if desired.

## How are these changes tested?
Existing UT for name change plus new UT for bug fix.

### CODE REVIEW

File: spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala [hunk 1]
```diff
@@ -950,28 +950,31 @@ class CometSparkSessionExtensions
...
   // various reasons) or Spark requests row-based output such as a `collect` call. Spark will adds
-  // another `ColumnarToRowExec` on top of `CometRowToColumnarExec`. In this case, the pair could
+  // another `ColumnarToRowExec` on top of `CometSparkToColumnarExec`. In this case, the pair could
   // be removed.
   case class EliminateRedundantTransitions(session: SparkSession) extends Rule[SparkPlan] {
     override def apply(plan: SparkPlan): SparkPlan = {
       val eliminatedPlan = plan transformUp {
-        case ColumnarToRowExec(rowToColumnar: CometRowToColumnarExec) => rowToColumnar.child
+        case ColumnarToRowExec(sparkToColumnar: CometSparkToColumnarExec) => sparkToColumnar.child
+        case CometSparkToColumnarExec(child: CometSparkToColumnarExec) => child
```
Comment (Kimahriman) [inline spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala:977]: This is the bug fix for duplicate nodes from AQE, I can make this a separate PR if needed

File: common/src/main/scala/org/apache/comet/CometConf.scala [hunk 2]
```diff
@@ -374,20 +374,22 @@ object CometConf extends ShimCometConf {
...
+    conf("spark.comet.sparkToColumnar.enabled")
       .internal()
-      .doc("Whether to enable row to columnar conversion in Comet. When this is turned on, " +
-        "Comet will convert row-based operators in " +
-        "`spark.comet.rowToColumnar.supportedOperatorList` into columnar based before processing.")
+      .doc(
+        "Whether to enable row to columnar conversion in Comet. When this is turned on, " +
+          "Comet will convert row-based operators in " +
+          "`spark.comet.sparkToColumnar.supportedOperatorList` into columnar based before " +
+          "processing.")
```
Comment (andygrove) [inline common/src/main/scala/org/apache/comet/CometConf.scala:384]: The documentation should also be updated (it still says this is for converting row to columnar only)

Comment (Kimahriman) [reply common/src/main/scala/org/apache/comet/CometConf.scala:384]: Good catch, only did the find replaces

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @Kimahriman

**Pull request merged**